pages/profile/profile.tsx

```diff
@@ -19,11 +24,34 @@ type profilePageScreenNavigationProp = NativeStackNavigationProp<
 
 type Props = {
     navigation: profilePageScreenNavigationProp;
+    userProfile: {};
 };
 
 const ProfilePage = ({ navigation }: Props) => {
 
-    //Code for profile pic upload
+    //Used for user info retrieval
+    const [user, loading, error]= useAuthState(firebaseAuth);
+    const [profileInfo, setProfileInfo]= React.useState({});
+
+    //Used to set user info
+    const [username, setUsername]= React.useState('');
+    const [email, setEmail]= React.useState('');
+    const [usernameModal, setUsernameModal]= React.useState(false);
+    const [emailModal, setEmailModal]= React.useState(false);
+
+    //Retrieves profile info when page first rendered
+    const getProfile= async () => {
+        const profileRef= doc(firestore, "users", user?.uid);
```

Comment:
Remember to handle the case where the document is not successfully fetched. Currently, it throws a render error but ideally we just want the profile page to appear to still be loading or appear empty